cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/adapter/PluginInstantiator.java

```diff
@@ -63,17 +65,26 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(PluginInstantiator.class);
 
+  private final LoadingCache<Location, ClassLoader> locationClassLoaders;
   private final LoadingCache<ArtifactDescriptor, ClassLoader> classLoaders;
   private final InstantiatorFactory instantiatorFactory;
   private final File tmpDir;
+  private final File pluginDir;
   private final ClassLoader parentClassLoader;
 
   public PluginInstantiator(CConfiguration cConf, ClassLoader parentClassLoader) {
     this.instantiatorFactory = new InstantiatorFactory(false);
-
     File tmpDir = new File(cConf.get(Constants.CFG_LOCAL_DATA_DIR),
       cConf.get(Constants.AppFabric.TEMP_DIR)).getAbsoluteFile();
+    File prefix = cConf.get(Constants.AppFabric.MRTASK_PLUGIN_DIR) != null ?
```

Comment:
This is used only for Mapper/Reducer tasks. Since in those tasks, the base directory for the PluginInstantiator will be the JAR name. That is, instead of data/namespaces/default/artifacts..... it will be artifacts.jar/data/namespaces/default/.... Hence we need to append the base directory name only for Mapper/Reducer. Hence the constant name.